signer/src/config/mod.rs

```diff
@@ -304,37 +295,21 @@ pub struct StacksNodeSettings {
     pub nakamoto_start_height: u64,
 }
 
-impl StacksSettings {
-    /// Create a new StacksSettings object by reading the relevant entries
-    /// in the signer's config.toml. The values there can be overridden by
-    /// environment variables.
-    ///
-    /// # Notes
-    ///
-    /// The relevant environment variables and the config entries that are
-    /// overridden are:
-    ///
-    /// * SIGNER_STACKS_API_ENDPOINT <-> stacks.api.endpoint
-    /// * SIGNER_STACKS_NODE_ENDPOINTS <-> stacks.node.endpoints
-    ///
-    /// Each of these overrides an entry in the signer's `config.toml`
-    pub fn new_from_config() -> Result<Self, Error> {
-        let source = File::with_name("./src/config/default");
-        let env = Environment::with_prefix("SIGNER")
-            .prefix_separator("_")
-            .list_separator(",")
-            .try_parsing(true)
-            .with_list_parse_key("stacks.node.endpoints")
-            .separator("_");
+impl Validatable for StacksConfig {
+    fn validate(&self, _: &Settings) -> Result<(), ConfigError> {
+        if self.endpoints.is_empty() {
+            return Err(ConfigError::Message(
+                "[stacks] Endpoints cannot be empty".to_string(),
+            ));
+        }
 
-        let conf = Config::builder()
-            .add_source(source)
-            .add_source(env)
-            .build()
-            .map_err(Error::SignerConfig)?;
+        if self.nakamoto_start_height == 0 {
+            return Err(ConfigError::Message(
+                "[stacks] Nakamoto start height must be greater than zero".to_string(),
+            ));
+        }
```

Comment:
So the Nakamoto start height can literally not be less than the number of blocks it takes to transition through the epochs to get to 3.0. It needs to transition from genesis -> 2.0 -> 2.0.5 -> 2.1 -> 2.2 -> 2.3 -> 2.4 -> 2.5 -> 3.0, so the actual minimum should be 9.

Reply:
> It needs to transition from (1.0) -> 2.0 -> 2.0.5 -> 2.1 -> 2.2 -> 2.3 -> 2.4 -> 2.5 -> 3.0, so the actual minimum should be 7 (1.0 & 2.0 can both be @ block 0 at least i know).

Oh good to know. Then maybe we should error if the start height is less than 7?

Reply:
Kinda feels fair to just make sure it's non-zero (for testing) -- in testnet/mainnet it will have to be the actual activation block heights which will be a lot higher... I wonder if there's a way to get this information from a nakamoto node (and skip it from config)...

Reply:
> I wonder if there's a way to get this information from a nakamoto node (and skip it from config)...

ooh that would be nice.